rcl/src/rcl/wait.c

```diff
@@ -75,29 +80,9 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   if (wait_set->subscriptions) {
-    rcl_ret_t ret = rcl_wait_set_resize_subscriptions(wait_set, 0);
+    rcl_ret_t ret = rcl_wait_set_resize(wait_set, 0, 0, 0, 0, 0);
```

Comment:
This change introduced a memory leak. If a wait set has no subscriptions neither of the other entities are being deleted. See #418 for a fix which updates the condition to be true if any kind of entity is present (not only subscriptions).